Dockerize the fids_frontend

Dogrock opened the pull request:
Place the newly separated fids_frontend components inside a nginx container so it can be more easily used for more than just Firestarter.

Commit:
Update nginx based container

BCK-5483

Commit:
Add compose, make and github actions

BCK-5483

Commit:
Rename git action workflow

BCK-5483

Commit:
Flesh out the readme

Commit:
Be consistent about end of file newline

NasaGeek commented on nginx/nginx.conf:
```diff
@@ -0,0 +1,24 @@
+        }
+
+        # airports, flights and positions uris are passed to backend
+        location ~ ^/(airports|flights|positions) {
```
To avoid having to make  further changes to this config as the backend API is extended, I think it'd make sense to put the backend endpoints under another component like `/api/`.

Dogrock replied:
Agreed, that makes future work much easier

NasaGeek commented on README.md:
```diff
@@ -0,0 +1,11 @@
+# FIDS Frontend - A sample frontend for FlightAware Data APIs
+
+## QuickStart
+Head on over to [FlightAware Firestarter](https://github.com/flightaware/firestarter) and follow the quickstart guide there. The documented commands will pull this FIDS Frontend as a ready to use Docker container that will present flight data from the Firestarer backend.
```
```suggestion
Head on over to [FlightAware Firestarter](https://github.com/flightaware/firestarter) and follow the quickstart guide there. The documented commands will pull this FIDS Frontend as a ready to use Docker container that will present flight data from the Firestarter backend.
```

NasaGeek commented on README.md:
```diff
@@ -0,0 +1,11 @@
+Head on over to [FlightAware Firestarter](https://github.com/flightaware/firestarter) and follow the quickstart guide there. The documented commands will pull this FIDS Frontend as a ready to use Docker container that will present flight data from the Firestarer backend.
+
+## fids-frontend
+FlightAware FIDS is a small webapp powered by a backend like the fids-backend service in FlightAware Firestarter. ou can use it to browse flight data by airport, presenting flights similarly to how you'd see them on a flight information display system (FIDS). Detailed information for individual flights can also be viewed.
```
```suggestion
FlightAware FIDS is a small webapp powered by a backend like the fids-backend service in FlightAware Firestarter. You can use it to browse flight data by airport, presenting flights similarly to how you'd see them on a flight information display system (FIDS). Detailed information for individual flights can also be viewed.
```

NasaGeek commented on README.md:
```diff
@@ -0,0 +1,11 @@
+## fids-frontend
+FlightAware FIDS is a small webapp powered by a backend like the fids-backend service in FlightAware Firestarter. ou can use it to browse flight data by airport, presenting flights similarly to how you'd see them on a flight information display system (FIDS). Detailed information for individual flights can also be viewed.
+
+This service is intended to be run in the context of a service like Firestarter. The nginx proxy configuration assumes there will be a locally avalible service known as "fids-backend". If running the fids-frontend as a stand-alone service the nginx proxy_pass configuration will need to changed to point to the location of your backend service.
```
```suggestion
This service is intended to be run in the context of a service like Firestarter. The nginx proxy configuration assumes there will be a locally available service at the "fids-backend" hostname. If running the fids-frontend as a stand-alone service the nginx proxy_pass configuration will need to changed to point to the location of your backend service.
```

NasaGeek commented on README.md:
```diff
@@ -0,0 +1,11 @@
+
+This service is intended to be run in the context of a service like Firestarter. The nginx proxy configuration assumes there will be a locally avalible service known as "fids-backend". If running the fids-frontend as a stand-alone service the nginx proxy_pass configuration will need to changed to point to the location of your backend service.
+
+Once running the service is accessable at http://localhost:8080 in your web browser (if not running Docker locally, use the Docker host's address).
```
```suggestion
Once running, the service is accessible at http://localhost:8080 in your web browser (if not running Docker locally, use the Docker host's address).
```

NasaGeek reviewed (commented):
Thanks for keeping the history. Really helps this project serve the dual purposes of demo and learning tool. I just had a couple of things to say directly about this PR, but then there's some other stuff that should be addressed that's already in master:

Previously, we had Flask injecting a Google Maps key into index.html to enable the Javascript to use it in static Google Maps queries. Now Flask isn't serving the webapp itself, we need some other way to grab those images. I'm thinking maybe we should add a new endpoint to the backend that just proxies the request back to Google Maps; then the key stays in the backend. You can just forward along all the query params (this sidesteps the issue we discussed with the raw Github image URLs, you'll just need to update them).

Commit:
Add new request to fetch maps api key from backend

Previously the Flask would inject the key into index.html. Since
the frondend is now separate this can no longer be done easily.
The api key remains with the rest of the fids_backend in Firestarter
and the frontend requests it from the backend.

Commit:
Move backend requests to all be under the /api resource to simplfy proxy rule

Commit:
Update README based on PR feedback

NasaGeek commented on client/src/components/FlightInfo/FlightInfo.js:
```diff
@@ -148,7 +164,7 @@ export default class FlightInfo extends Component {
             }
 
-            return <div className="d-flex align-items-center"><img src={`https://maps.googleapis.com/maps/api/staticmap?size=640x400&markers=anchor:center|icon:https://github.com/flightaware/firestarter/raw/master/fids/images/aircraft_${image_bearing}.png|${positions[0].latitude},${positions[0].longitude}&path=color:0x0000ff|weight:5${latlon}&key=${window.GOOGLE_MAPS_API_KEY}`}/></div>
+            return <div className="d-flex align-items-center"><img src={`https://maps.googleapis.com/maps/api/staticmap?size=640x400&markers=anchor:center|icon:https://github.com/flightaware/firestarter/raw/master/fids/images/aircraft_${image_bearing}.png|${positions[0].latitude},${positions[0].longitude}&path=color:0x0000ff|weight:5${latlon}&key=${google_maps_api_key}`}/></div>
```
There's still the problem of the aircraft image URLs. I think just storing them in this repo and updating the URLs accordingly is acceptable.

Dogrock replied:
This was supposed to already be updated to point at this repo, missed copy/paste error.

NasaGeek commented on nginx/nginx.conf:
```diff
@@ -0,0 +1,24 @@
+        }
+
+        # airports, flights and positions uris are passed to backend
+        location ~ ^/(api) {
```
```suggestion
        location /api {
```
Cleaner I think

NasaGeek commented on nginx/nginx.conf:
```diff
@@ -0,0 +1,24 @@
+
+        # airports, flights and positions uris are passed to backend
+        location ~ ^/(api) {
+            proxy_pass http://fids-backend:5000;
```
```suggestion
            proxy_pass http://fids-backend:5000/;
```
I could be wrong, but I _think_ that by specifying this trailing slash, you can remove all the `/api/` prefixes you had to add to the routes in the backend. I think it's cleaner that way.

If this doesn't work like I thought, there is likely some other way to accomplish this.

Dogrock replied:
Yeah, this works as expected and makes the backend much cleaner

NasaGeek commented on README.md:
```diff
@@ -0,0 +1,11 @@
+## fids-frontend
+FlightAware FIDS is a small webapp powered by a backend like the fids-backend service in FlightAware Firestarter. You can use it to browse flight data by airport, presenting flights similarly to how you'd see them on a flight information display system (FIDS). Detailed information for individual flights can also be viewed.
+
+This service is intended to be run in the context of a service like Firestarter. The nginx proxy configuration assumes there will be a locally available service at the "fids-backend" hostname. If running the fids-frontend as a stand-alone service the nginx proxy_pass configuration will need to changed to point to the location of your backend service.
```
```suggestion
This service is intended to be run in the context of a service like Firestarter. The nginx proxy configuration assumes there will be a locally available service at the "fids-backend" hostname. If running the fids-frontend as a stand-alone service the nginx proxy_pass configuration will need to be changed to point to the location of your backend service.
```

Commit:
Tune api proxy statement

* removes the need to prefix backend endpoints with /api

Commit:
Update map icon image source url

* source from the local repository since they've been moved too

Dogrock committed:
Update README.md

Co-authored-by: Chris Roberts <NasaGeek@users.noreply.github.com>
